src/views/PuzzleCollection/PuzzleCollection.tsx

```diff
@@ -23,14 +23,15 @@ import { get, del, put, abort_requests_in_flight } from "requests";
 import { SortablePuzzleList } from "./SortablePuzzleList";
 import { openACLModal } from "ACLModal";
 import swal from "sweetalert2";
+import { RouteComponentProps } from "react-router-dom";
 
 export function PuzzleCollection({
     match: {
         params: { collection_id },
     },
-}: {
-    match: { params: { collection_id: number } };
-}): JSX.Element {
+}: RouteComponentProps<{
+    collection_id: any; // string treated as a number
```

Comment:
That's always going to be `string` I think, and we should `parseInt` to convert it to a number if we need a number I reckon?

Reply:
Yep we should, but we don't currently.  I was trying to keep functional changes separate from type changes, but it's simple enough that I can make those updates in this PR.

The thing I need to investigate further is whether is whether conversion to number is necessary at all.  For example, `collection_id` gets passed into `ACLModal`, who says it should be a `number` (that's why TS complains about string).  But then I look at `ACLModal` and it just formats the number back into a string for a URL.  So is the right answer to change `ACLModal`'s `props.puzzle_collection_id` into a `string|number` instead of converting with `parseInt()`?